Fix post-0.50.0 delete metrics test

## Current Behavior
Red.
https://rest.fs.neo.org/Aku38wpn2xEqfEPxmwyAa7N3VnAkMV8PEVkpLqwKM1z3/4498-1765389808/index.html#suites/baabcbd9a1d15d3e8b09a2606c8d73a8/e2e518285797fda1/

test_sn_ir_metrics fails with
> AssertionError: invalid value for neofs_node_engine_delete_time_bucket
assert 0 &gt;= 1
 +  where 0 = len([])

## Expected Behavior
Green.

## Possible Solution
Change to neofs_node_engine_drop_time_bucket for node >0.50.2.

## Steps to Reproduce
Run testcases against master node.

## Context
https://github.com/nspcc-dev/neofs-node/pull/3727 introduced a different (drop) metric for API used by control service.

## Regression
Not exactly.

--- a/pytest_tests/tests/metrics/test_sn_metrics.py
+++ b/pytest_tests/tests/metrics/test_sn_metrics.py
@@ -28,6 +28,7 @@ from helpers.rest_gate import (
     get_via_rest_gate,
     upload_via_rest_gate,
 )
+from helpers.utility import parse_version
 from helpers.wallet_helpers import create_wallet
 from helpers.wellknown_acl import PUBLIC_ACL
 from neofs_testlib.env.env import NeoFSEnv, NodeWallet
@@ -269,7 +270,13 @@ def test_sn_ir_metrics(neofs_env_single_sn: NeoFSEnv, default_wallet: NodeWallet
         after_metrics_sn = get_metrics(sn)
         allure.attach(json.dumps(dict(after_metrics_sn)), "sn metrics object drop", allure.attachment_type.JSON)
 
-    for metric in ("neofs_node_engine_delete_time_bucket",):
+    deletion_metric = "neofs_node_engine_delete_time_bucket"
+    if parse_version(neofs_env_single_sn.get_binary_version(neofs_env_single_sn.neofs_node_path)) > parse_version(
+        "0.50.2"
+    ):
+        deletion_metric = "neofs_node_engine_drop_time_bucket"
+
+    for metric in (deletion_metric,):
         assert len([m for _, m in enumerate(after_metrics_sn[metric]) if m["value"] >= 1]) >= 1, (
             f"invalid value for {metric}"
         )

